scripts/public_api_report.py

```diff
@@ -489,6 +503,287 @@ class CodeReferenceFilter:
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/glob_reader_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/manual_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/query_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/s3_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/s3_subdir_reader_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/subdir_reader_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/batch_kwargs_generator/table_batch_kwargs_generator.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="ValidationOperators are now run from Checkpoints: https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide#manually-migrate-v2-checkpoints-to-v3-checkpoints",
+            filepath=pathlib.Path(
+                "great_expectations/validation_operators/types/validation_operator_result.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="ValidationOperators are now run from Checkpoints: https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide#manually-migrate-v2-checkpoints-to-v3-checkpoints",
+            filepath=pathlib.Path(
+                "great_expectations/validation_operators/validation_operators.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="ValidationActions are now run from Checkpoints: https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide#manually-migrate-v2-checkpoints-to-v3-checkpoints",
+            name="run",
+            filepath=pathlib.Path("great_expectations/checkpoint/actions.py"),
+        ),
```

Comment:
Just to confirm, I would think we would consider the `_run` methods of individual Actions to be public, since those can be overridden using plugins, and are very user-facing. Is that compatible with this?

Reply:
What if the classes of the actions themselves are part of the public API? e.g. SlackNotificationAction, PagerdutyAlertAction, etc? That way the docstring explaining how to configure the action is documented as part of API docs?

Reply:
I think that should work!

Reply:
Ah I see also your point about overriding. Yes it would be good to also document the `_run` method of the abstract class along with the abstract class (ValidationAction) with a description about how to build your own action.

Reply:
Upcoming PR: https://github.com/great-expectations/great_expectations/pull/6803